src/rpc/core_rpc_server.cpp

```diff
@@ -887,9 +887,9 @@ namespace cryptonote
       return false;
     }
 
-    cryptonote::account_public_address acc = AUTO_VAL_INIT(acc);
+    cryptonote::address_parse_info info;
 
-    if(!req.wallet_address.size() || !cryptonote::get_account_address_from_str(acc, m_testnet, req.wallet_address))
+    if(!req.wallet_address.size() || !cryptonote::get_account_address_from_str(info, m_testnet, req.wallet_address) || info.is_subaddress)
     {
       error_resp.code = CORE_RPC_ERROR_CODE_WRONG_WALLET_ADDRESS;
       error_resp.message = "Failed to parse wallet address";
```

Comment:
New ones are much better, especially as in this case it's just a new define to add, unlike the wallet where you have to add a whole new class. It's always annoying when you use an API and it just tells you "failed, but you have to guess why".